modules/dnn/src/net_impl.cpp

```diff
@@ -469,8 +469,9 @@ void Net::Impl::allocateLayer(int lid, const LayersShapesMap& layersShapes)
     for (std::set<int>::const_iterator i = ld.inputLayersId.begin(); i != ld.inputLayersId.end(); i++)
         allocateLayer(*i, layersShapes);
 
-    // bind inputs
-    if (ld.id == 0 && netInputLayer->supportBackend(preferableBackend))  // DataLayer
+    // bind inputs for DataLayer
+    if (ld.id == 0 &&
+        (netInputLayer->supportBackend(preferableBackend) || preferableBackend == DNN_BACKEND_INFERENCE_ENGINE_NGRAPH))
```

Comment:
Can you check why `netInputLayer` returns false in case of `DNN_BACKEND_INFERENCE_ENGINE_NGRAPH`?

Reply:
I think it's better, let's move